sklearn/svm/tests/test_svm.py

```diff
@@ -439,6 +439,76 @@ def test_sample_weights():
     assert_array_almost_equal(dual_coef_no_weight, clf.dual_coef_)
 
 
+@pytest.mark.parametrize(
+    "CLF, params",
+    [(svm.SVC,
+      {"err_msg":
+       'Invalid input - all samples have zero/negative weights'}),
+     (svm.NuSVC,
+     {"err_msg":
+      '(negative dimensions are not allowed|nu is infeasible)'}),
+     (svm.SVR,
+     {"err_msg":
+      'Invalid input - all samples have zero/negative weights'}),
+     (svm.NuSVR,
+     {"err_msg":
+      'Invalid input - all samples have zero/negative weights'}),
+     (svm.OneClassSVM,
+     {"err_msg":
+      'Invalid input - all samples have zero/negative weights'})
+     ],
+    ids=['SVC', 'NuSVC', 'SVR', 'NuSVR', 'OneClassSVM']
+)
+@pytest.mark.parametrize(
+    "sample_weight",
+    [[0] * len(Y), [-0.3] * len(Y)],
+    ids=['weights-are-zero', 'weights-are-negative']
+)
+def test_negative_sample_weights_mask_all_samples(CLF, params, sample_weight):
+    clf = CLF(kernel='linear')
+    with pytest.raises(ValueError, match=params['err_msg']):
+        clf.fit(X, Y, sample_weight=sample_weight)
+
+
+@pytest.mark.parametrize(
+    "CLF, params",
+    [(svm.SVC, {"err_msg":
+     'Invalid input - all samples with positive weights have the same label'}),
+     (svm.NuSVC, {"err_msg": 'specified nu is infeasible'})],
+    ids=['SVC', 'NuSVC']
+)
+@pytest.mark.parametrize(
+    "sample_weight",
+    [[0, -0.5, 0, 1, 1, 1],
+     [1, 1, 1, 0, -0.1, -0.3]],
+    ids=['mask-label-1', 'mask-label-2']
+)
+def test_negative_weights_svc_leave_just_one_label(CLF, params, sample_weight):
+    clf = CLF(kernel='linear')
+    with pytest.raises(ValueError, match=params['err_msg']):
+        clf.fit(X, Y, sample_weight=sample_weight)
+
+
+@pytest.mark.parametrize(
+    "CLF, model",
+    [(svm.SVC, {'when-left': [0.3998,  0.4], 'when-right': [0.4,  0.3999]}),
+     (svm.NuSVC, {'when-left': [0.3333,  0.3333],
+      'when-right': [0.3333, 0.3333]})],
+    ids=['SVC', 'NuSVC']
+)
+@pytest.mark.parametrize(
+    "sample_weight, mask_side",
+    [([1, -0.5, 1, 1, 1, 1], 'when-left'),
+     ([1, 1, 1, 0, 1, 1], 'when-right')],
+    ids=['partial-mask-label-1', 'partial-mask-label-2']
+)
+def test_negative_weights_svc_leave_two_labels(CLF, model,
+                                               sample_weight, mask_side):
+    clf = CLF(kernel='linear')
+    clf.fit(X, Y, sample_weight=sample_weight)
+    assert_array_almost_equal(clf.coef_, [model[mask_side]], decimal=4)
```

Comment:
you should use `assert_allclose` instead.
Because of the fitting, I think that what you want to test is that the 2 coefs are the same:

```python
coef = np.abs(estimator.coef_).ravel()
assert coef[0] == pytest.approx(coef[1], rel=1e-3)
```

Reply:
Also it should work for the regressor:

cf the following test:
https://github.com/glemaitre/scikit-learn/commit/5b30266954f30ed354160c22824e983d832b1105#diff-a286b98dc7871c4de856c76a20cb57cdR499

Reply:
I did both tests: 
a. assert_allclose between the coeff_ and expected value
b. approx between coeff_[0] and coeff[1]